walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,210 @@
+      }
+    }
+
+    return suitableSlots;
+  }
+
+  private TimeTablePair registerRelevantEvents(final Collection<Event> events,
+                                               final Collection<String> mandatoryAttendees,
+                                               final Collection<String> optionalAttendees) {
+    /**
+     * The timeTable collection holds TimeRanges which represent empty time slots for
+     * all the people invited.
+     */
+    List<TimeRange> timeTable = new ArrayList<>();
+    List<TimeRange> optionalTimeTable = new ArrayList<>();
+    timeTable.add(TimeRange.WHOLE_DAY);
+    optionalTimeTable.add(TimeRange.WHOLE_DAY);
+
+    for (Event event : events) {
+      if (relevantEvent(event, mandatoryAttendees)) {
+        splitTimeTable(timeTable, event.getWhen());
+        splitTimeTable(optionalTimeTable, event.getWhen());
+      }
+
+      if (relevantEvent(event, optionalAttendees)) {
+        splitTimeTable(optionalTimeTable, event.getWhen());
+      }
+    }
+
+    return new TimeTablePair(timeTable, optionalTimeTable);
+  }
+
+  private Boolean relevantEvent(final Event event, final Collection<String> relevantPeople) {
+    for (String person : relevantPeople) {
+      if (event.getAttendees().contains(person)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Possible cases:
+   * 1. originalTimeTable:  [--------------------------]
+   *    splitter:                     [-----]
+   *    result:             [---------]     [----------]
+   *
+   * 2. originalTimeTable:  [------]   ...   [---------]
+   *    splitter:                [-------------]
+   *    result:             [----]             [-------]
+   *
+   * 3. originalTimeTable:  [------]   ...  [----------]
+   *    splitter:                     [-----]
+   *    result:             [------]        [----------]
+   *
+   * 4. originalTimeTable:  [------]           [-------]
+   *    splitter:                     [-----]
+   *    result:             [------]           [-------]
+   *
+   * Mention: for this method, it will only matter if the splitter is contained or not.
+   */
+  private void splitTimeTable(final List<TimeRange> originalTimeTable,
+                              final TimeRange timeSplitter) {
+    if (originalTimeTable.isEmpty()) {
+      return;
+    }
+
+    /**
+     * Find the element with the largest starting point <= the starting point
+     * of @param timeSplitter. This will determine in which case the program is in.
+     */
+    int firstRelevantTimeslotIdx = TimeRange.lowerBound(originalTimeTable, timeSplitter);
+
+    if (originalTimeTable.get(firstRelevantTimeslotIdx).contains(timeSplitter)) {  // Case 1
+      splitAtTimeslot(originalTimeTable, timeSplitter, firstRelevantTimeslotIdx);
+    } else {  // Case 2, 3, 4
+      modifyOverlappingSlots(originalTimeTable, timeSplitter, firstRelevantTimeslotIdx);
+    }
+  }
+
+  /**
+   * For case 1, there could be 4 variations:
```

Comment:
This is referring to case 1 from the other method's javadoc, which is confusing. Consider adding the example of what case 1 looks like in this comment.